src/irmin/tree.ml

```diff
@@ -785,6 +779,7 @@ module Make (P : Private.S) = struct
         acc ->
         acc Lwt.t =
      fun ~force ~uniq ~pre ~post ~path ?depth ~node ~contents ~tree t acc ->
+      let cache = force = `True in
```

Comment:
Now that I read my code again it seems I was too quick when implementing this behaviour. I see 4 cases for this `force` parameter and not 3:
- Force and cache: `` `True ``
- Force and no cache: `` `And_clear ``
- No force and cache (this may happen when computing the hashes for `uniq`): `` `False `` on master (I'll adapt my PR to keep that behaviour)
- No force and no cache

I would change the fold API
```ml
(* from *)
?force:[ `True | `False of key -> 'a -> 'a Lwt.t | `And_clear ] ->
```

```ml
(* to *)
?force:bool ->
?cache:bool ->
?skip:key -> 'a -> 'a Lwt.t ->
```

Reply:
It might be a shame to lose the type info that the `skip` function is called only when `~force:false` (and ignored otherwise). What about this?

```ocaml
?force:[ `True | `False of key -> 'a -> 'a Lwt.t ] -> ?cache:bool ->
```

Reply:
Agreed. 

Should I include that change in this PR?

Reply:
I think it makes sense in this PR, but you're welcome to split it if you prefer.

Reply:
Done